src/main/java/org/fogbowcloud/sebal/parsers/WeatherStation.java

```diff
@@ -25,112 +22,58 @@
 		this(new Properties());
 	}
 
-	public WeatherStation(Properties properties) throws URISyntaxException,
-			HttpException, IOException {
+	public WeatherStation(Properties properties)
+			throws URISyntaxException, HttpException, IOException {
+		this(properties, new FTPStationOperator(properties));
+	}
+
+	protected WeatherStation(Properties properties, FTPStationOperator stationOperator) {
 		this.properties = properties;
-		this.stationOperator = new FTPStationOperator(properties);
+		this.stationOperator = stationOperator;
 	}
 
-	public void persistStations() throws IOException {
-		JSONArray stations = new JSONArray();
-		for (String line : IOUtils.readLines(new FileInputStream(
-				"stations.html"))) {
-			if (line.contains("var html")) {
-				String[] split = line.split("<br />");
-				JSONObject json = new JSONObject();
-				String split0 = split[0];
-				String split00 = split0
-						.substring(split[0].lastIndexOf(":") + 1,
-								split[0].indexOf("</b>")).trim();
-
-				json.put("id", split00.split("-")[0].trim());
-				json.put("name", split00.substring(split00.indexOf("-") + 1)
-						.trim());
-				json.put(
-						"lat",
-						Double.parseDouble(split[1].substring(
-								split[1].indexOf(":") + 1).trim()));
-				json.put(
-						"lon",
-						Double.parseDouble(split[2].substring(
-								split[2].indexOf(":") + 1).trim()));
-				json.put(
-						"altitude",
-						Double.parseDouble(split[3]
-								.substring(split[3].indexOf(":") + 1)
-								.replaceAll("m", "").trim()));
-				stations.put(json);
-			}
+	public String getStationData(double lat, double lon, Date date) {
+		LOGGER.debug("latitude: " + lat + " longitude: " + lon + " date: " + date);
+
+		int daysWindow = 0;
+		List<JSONObject> nearStations = stationOperator.findNearestStation(date, lat, lon,
+				daysWindow);
+
+		if (nearStations != null) {
+			return selectStationRecords(date, nearStations, daysWindow);
 		}
-		IOUtils.write(stations.toString(2), new FileOutputStream(
-				"stations.json"));
+
+		return null;
 	}
 
-	protected String readFullRecord(Date date, List<JSONObject> stations, int numberOfDays) {
-		LOGGER.info("Near stations found...reading full record");
+	protected String selectStationRecords(Date date, List<JSONObject> stations, int numberOfDays) {
+		LOGGER.info("Near stations found... reading and selecting full record");
 		Date begindate = new Date(date.getTime() - numberOfDays * StationOperatorConstants.A_DAY);
 		Date endDate = new Date(date.getTime() + numberOfDays * StationOperatorConstants.A_DAY);
 
-		if (!stations.isEmpty()) {
+		if (stations != null && !stations.isEmpty()) {
 			LOGGER.debug("beginDate: " + begindate + " endDate: " + endDate);
+
 			for (JSONObject station : stations) {
 				try {
 					JSONArray stationData = stationOperator.readStation(station.optString("id"),
 							StationOperatorConstants.DATE_FORMAT.format(begindate),
 							StationOperatorConstants.DATE_FORMAT.format(endDate));
 
```

Comment:
found

Reply:
I don't understand the change that you want!